qutip/solver/integrator/integrator.py

```diff
@@ -67,12 +67,12 @@ class Integrator:
     name = None
     method = ""
 
-    def __init__(self, system, options):
-        self.system = system
+    def __init__(self, solver):
```

Comment:
Replacing `system, options` with `solver` means that now the interface between integrators and solvers is the entire solver and all its behaviour.  To some extent this is the point of this change -- i.e. for integrators to be able to tailor themselves to particular solvers -- but I'm concerned that there is zero indication (via documentation in docstrings or otherwise) exactly what behaviour integrators can depend on.

How can we make explicit what parts of solvers integrators may access and which parts specific integrators do access?